src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Wrong sides');
+    }
 
-export class Circle {
+    if (a >= b + c || b >= a + c || c >= b + a) {
+      throw new Error('Wrong sides');
+    }
+  }
 
+  getArea(): number {
+    const { a, b, c } = this;
+
+    const p = (a + b + c) / 2;
+
+    return Math.floor(Math.sqrt(p * (p - a) * (p - b) * (p - c)) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Wrong radius');
+    }
+  }
 
+  getArea(): number {
+    const { radius } = this;
+
+    return Math.floor(Math.PI * radius * radius * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Wrong sides');
```

Comment:
let us create separate function for `throw error` to prevent duplicating. Pass phrase you want to be thrown to this function